packages/zoe/src/contractFacet/rightsConservation.js

```diff
@@ -37,24 +37,43 @@ const sumByBrand = amounts => {
  * indexed by issuer
  */
 const assertEqualPerBrand = (leftSumsByBrand, rightSumsByBrand) => {
-  const leftKeys = leftSumsByBrand.keys();
-  const rightKeys = rightSumsByBrand.keys();
-  assert.equal(
-    leftKeys.length,
-    rightKeys.length,
-    X`${leftKeys.length} should be equal to ${rightKeys.length}`,
-  );
-  leftSumsByBrand
-    .keys()
-    .forEach(brand =>
+  // We cannot assume that all of the brand keys present in
+  // leftSumsByBrand are also present in rightSumsByBrand. A empty
+  // amount could be introduced or dropped, and this should still be
+  // deemed "equal" from the perspective of rights conservation.
+
+  // Thus, we should allow for a brand to be missing from a map, but
+  // only if the sum for the brand in the other map is empty.
+
+  /**
+   * A helper that either gets the sum for the specified brand, or if
+   * the brand is absent in the map, returns an empty amount.
+   *
+   * @param {Store<Brand, Amount>} sumsByBrandMap
+   * @param {Brand} brand
+   * @param {Amount} amount
+   * @returns {Amount}
+   */
+  const getOrEmpty = (sumsByBrandMap, brand, amount) => {
+    if (!sumsByBrandMap.has(brand)) {
+      return amountMath.makeEmptyFromAmount(amount);
+    }
+    return sumsByBrandMap.get(brand);
+  };
+
+  const assertSumsEqualInMap = (mapToIterate, mapToCheck) => {
+    mapToIterate.keys().forEach(brand => {
```

Comment:
If you union the brands, you can do each comparison only once.
```suggestion
    const allBrands = new Set([...mapToIterate.keys(), ...mapToCheck.keys()]);
    allBrands.forEach(brand => {
```

Reply:
Oo, nice!

Reply:
huh, looking at this method in isolation though, we don't know if the brand is defined in the left or the right. And to make empty, we need an amount from the other map. The code gets messy because it doesn't have a place to stand. Let me see if I can improve this.

Reply:
Ok, pushed up a version that uses allBrands, but is a little more verbose